### ISSUE TITLE
QueueTrigger Function not respecting Singleton Attribute

### ISSUE THREAD

**Issue by aboersch** [Azure/Azure-Functions#422]
In a local Development environment using VS2017 Tooling and func CLI the QueueTrigger is not respecting the SingletonAttribute to prevent multiple queue items being processed in parallel.

Repo: https://github.com/aboersch/SingletonQueueFunctionApp

It also appears there is no lock file being created by the Consuming Function, only by the producer.

Windows 10.0.15063 Build 15063
Visual 2017 15.3.0 Preview 7.0
Azure Storage Emulator 5.1
Azure Function Tools for VS2017 0.2

**Comment by lindydonna** [comment]
Yes, SingletonAttribute is not supported in Azure Functions. We should add a warning to explain this.

**Comment by cmeiklejohn** [comment]
Is this still the case?  What's odd is the following:

* The singleton attribute appears to work correctly in the simulator locally.
* When deployed, it doesn't work and I have multiple invocations running at the same time.  However, I *do* see messages in the Application Insights output for the function that shows the singleton lock being acquired.  What do these messages mean?

**Comment by warpdriv** [comment]
This would be nice to have from several aspects especially for those (like me) that need queue consumption in the order the message was put in. Example: processing orders.

Thank you
